Move to Firebase hosting. Remove server code.

Splaktar opened the pull request:
- UrlShortener and analytics server features removed.
- Configure for Firebase Hosting and HTTPS.
- Change routing to use `gdg.events/android/events` instead of `android.gdg.events`.
- Allow location API to work in Chrome 50+ once again.

Please check it out here: https://firefly-1c312.firebaseapp.com/ and https://gdg.events/

Fixes #26, #58, #97, #80, #81.
Closes #82, #101, #96.

tasomaniac commented on client/app/main/main.html:
```diff
@@ -11,14 +11,14 @@
-          <md-list-item class="md-2-line" ng-repeat="event in allEvents | orderBy:distanceFromHere | limitTo:5"
-                        ng-click="openEvent(event._id)">
+          <md-list-item class="md-2-line" ng-repeat="event in vm.nearEvents | limitTo:5"
+                        ng-click="vm.openEvent(event._id)">
```
Noob question. What does vm stand for?

Splaktar replied:
ViewModel or Controller. Angular 1.x style guide uses `vm` and Angular 2 style guide has switched to `$ctrl`.

tasomaniac commented on server/views/404.html:
```diff
@@ -1,157 +0,0 @@
-<!DOCTYPE html>
```
I wonder if this 404 page was used. Right now if you put a random I'd, it opens the page with empty cards. 404 page with message would be nicer. Just checked, the old site redirects to main page.

Splaktar replied:
Thanks, fixed in `shorturlEvent.controller.js`.

tasomaniac commented:
I wonder what do we think about the subdomain? Does Firebase provide ssl for subdomains? If so, we can keep the old url structure right?

Splaktar commented:
@tasomaniac I don't think that Firebase supports HTTPS for subdomains unless you want to deploy a separate Firebase for each subdomain. That would be a nightmare to maintain though.

Splaktar commented:
I'm seeing some funky behavior on the `md-virtual-repeat`s on the specific tag pages like: https://gdg.events/devfest/events

The first 3 elements are repeating at the end and the sorting doesn't seem right. I'll try to look into it later.

tasomaniac commented:
Looks good to me. You can merge it once you fixed the problem.

Splaktar commented:
OK, fixed the odd behavior of repeating elements in md-virtual-repeat.

Also fixed

- Clicking on the link in map markers didn't work
- Tagged event page was loading events from all tags instead of just it's own tag
- Updated tests and readme

Splaktar commented:
The only remaining issue that I'm seeing is related to the Hub. Sometimes I'm not getting back all of the results that I would expect querying for a tag like DevFest for upcoming events. I'm investigating that separate from this PR.

tasomaniac commented:
:+1: Great work.

Splaktar commented:
This looks like it will save about $70 per month on GCP hosting costs. Perhaps more when counting static IP and load balancer, etc.
